api/mesh/v1alpha1/health_check.proto

```diff
@@ -39,6 +40,58 @@ message HealthCheck {
 
     // Number of consecutive healthy checks before considering a host healthy.
     uint32 healthy_threshold = 4 [ (validate.rules).uint32 = {gt : 0} ];
+
+    // Tcp defines optional configuration for specifying bytes to send and
+    // expected response during the health check
+    message Tcp {
+      // Bytes which will be send during the health check to the target
+      google.protobuf.BytesValue send = 1;
+
+      // Bytes blocks expected as a response. When checking the response,
+      // “fuzzy” matching is performed such that each block must be found, and
+      // in the order specified, but not necessarily contiguous.
+      repeated google.protobuf.BytesValue receive = 2;
+    }
+
+    Tcp tcp = 5;
```

Comment:
use `oneof` feature of Protobuf.

Reply:
I was thinking about it and that was my initial idea, but we would have to nest the `http`/`tcp` blocks in other property (something like `protocol_config`). It definitely makes more logical sense to have this validation on the level of protobuf schemas (especially if in the future we would like to introduce some other types of health checking). @nickolaev @jakubdyszkiewicz do you think one more property level is acceptable cost for us? Or maybe there is some protobuf property I couldn't find which would remove the need of separate property and let us us the blocks without nesting them?

Reply:
hmm, I think I'm wrong here, let me check

Reply:
Implemented change according to your suggestion :-)